### ISSUE TITLE
[Bug] Tesseract not properly updating current status to reflect adjacent block status

### ISSUE THREAD

**Issue by drivetheory** [SuperMartijn642/Tesseract#58]
**Version Info**
- Minecraft: 
FTB Direwolf20 1.18.2 v1.9
Forge 40.1.80

- Tesseract v1.027
- Applied Energistics 2 v11.1.5
- Mekanism v10.2.5
- (other mods may also be affected but are untested at this time)

**Description of the Bug**
Tesseract stops working properly

**Steps to Reproduce**
01) place Tesseract 1 (Tes1), configure to send only, create channel to send energy, activate,
02) place Tesseract 2 (Tes2), configure to receive only, select created energy channel, activate
03) observe Tes2 fill with energy remotely from Tes1
04) place 1 Mekanism Universal Cable (MUC) adjacent to Tes2,
05) configure MUC to pull mode so that it extracts energy from Tes2
06) Tes2 fills MUC with energy
07) observe Tes2 is now empty and is no longer being filled with energy remotely from Tes1
08) place 1 Applied Energestics 2 (AE2) Energy Acceptor on other side of MUC.
09) place 1 AE2 Energy Cell adjacent to AE2 Energy Acceptor
10) observe MUC is now completely empty
11) observe Tes2 is still empty

**Screenshot Set A**
![2022-11-12_23 05 11](https://user-images.githubusercontent.com/1679773/201505763-f03d95ea-2b3a-4f52-a473-d2230b10eb84.png)
![2022-11-12_23 06 43](https://user-images.githubusercontent.com/1679773/201505769-ea35d7bb-bf54-42e4-b45c-c0c9313bed9f.png)
![2022-11-12_23 06 56](https://user-images.githubusercontent.com/1679773/201505776-b8cd0077-2168-4b40-afd1-2427175e2519.png)
![2022-11-12_23 08 51](https://user-images.githubusercontent.com/1679773/201505809-632f7ba1-8877-4130-8b51-0f81445fd9b8.png)
![2022-11-12_23 22 59](https://user-images.githubusercontent.com/1679773/201505816-c81f8c2d-00a5-4250-b85f-b56b95f7073a.png)

**Steps to create alternate outcome due to change in order of block placement operations**
01) place Tesseract 1 (Tes1), configure to send only, create channel to send energy, activate,
02) place Tesseract 2 (Tes2), configure to receive only, select created energy channel, activate
03) place 1 Mekanism Universal Cable (MUC)
04) place 1 Applied Energestics 2 (AE2) Energy Acceptor adjacent to MUC.
05) place 1 AE2 Energy Cell adjacent to Energy Acceptor
06) configure MUC to pull mode so that it extracts energy from Tes2
07) observe MUC stays filled with energy due to AE2 Energy Cell being filled with energy
08) observe Tes2 is continuously filled with energy being sent from Tes1

**Screenshot Set B**
![2022-11-12_23 24 04](https://user-images.githubusercontent.com/1679773/201505826-3b98becc-a310-470d-8016-d151f3e49f3d.png)
![2022-11-12_23 24 15](https://user-images.githubusercontent.com/1679773/201505828-6dca7789-14db-4373-9343-39eb47e2cb82.png)

**Comment by SuperMartijn642** [comment]
Don't know why, but I completely missed this issue report, so sorry for the long wait...

The tesseracts themselves don't store anything.
The One Probe shows data from the block's energy capability. The capability allows other things to interact with the block in terms of energy, so transferring and observing. In most cases this will show energy stored in the block itself, however the tesseracts allow blocks to directly transfer energy to blocks around other tesseracts on the same channel. This means the tooltip will, somewhat confusingly, show the sum of all energy stored in the blocks around tesseracts on the same channel as the tesseract you're looking at.
The same holds for items and fluids.

I have now added integration with Jade and The One Probe in version 1.0.29. The integration removes the energy and fluid tooltips to avoid confusion.

**Comment by drivetheory** [comment]
I don't think you actually understood what I was demonstrating.....

The tool tip matters nothing. the AE2 blocks were NOT RECEIVING ENERGY from/through the tesseract in the first example.

**Comment by drivetheory** [comment]
FIRST EXAMPLE LISTED ABOVE, NO ENERGY PASSING THROUGH TESSERACT
![no energy](https://user-images.githubusercontent.com/1679773/220051877-c1f55f4b-ab50-4946-bf01-0cf642c7aa2e.jpg)


ATERNATE BLOCK PLACEMENT ORDER, ENERGY PASES THROUGH.....
![energy](https://user-images.githubusercontent.com/1679773/220052015-14e6e166-d440-4618-9003-5bf29a5e9dd2.jpg)


you're prematurely closing this thread...

**Comment by SuperMartijn642** [comment]
No need to spam in all caps.

Your issue is named `Tesseract not properly updating current status to reflect adjacent block status` and in your description, the thing you mention is the tesseract not 'being filled' with energy and the tesseract being empty. You don't mention anything about placement order or it not transferring energy to the cable while you expect it to, just that the tesseract itself shows it is empty.
As such, I explained that the tesseract does not store any energy and added integration with The One Probe to remove the energy tooltip and avoid such confusion.

What you are describing now sounds like a duplicate of #49, #56 or #57.

**Comment by drivetheory** [comment]
Oh so the second half of my original post where i explicitly described a "change in order of block placement operations" affecting outcome does not mention placement order. 

LOL. 👍 

re: #49/56/57 
no, i literally gave you a reporducable step by step procedure to recreate the problem, 
AND also defined specifically how the order of block placement caused the problem.
and provided photos in all accounts...

**Comment by drivetheory** [comment]
also, my sincerest apologies for using the word "status" in the threat title instead of saying "the tesseract stops checking to see if there are any connected blocks on any sides that it needs to output to after it checks only once the moment it is initially placed"
